#! /bin/sh /usr/share/dpatch/dpatch-run
## 21_avoid_ioseek.dpatch by Daigo Moriwaki <daigo@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: A work around for Bug#406388.

@DPATCH@
diff -urNad trunk~/lib/rubygems/package/tar_reader.rb trunk/lib/rubygems/package/tar_reader.rb
--- trunk~/lib/rubygems/package/tar_reader.rb	2008-02-29 17:55:01.000000000 +0900
+++ trunk/lib/rubygems/package/tar_reader.rb	2008-05-10 15:44:05.000000000 +0900
@@ -47,10 +47,10 @@
 
       skip = (512 - (size % 512)) % 512
 
-      if @io.respond_to? :seek then
-        # avoid reading...
-        @io.seek(size - entry.bytes_read, IO::SEEK_CUR)
-      else
+      #if @io.respond_to? :seek then
+      #  # avoid reading...
+      #  @io.seek(size - entry.bytes_read, IO::SEEK_CUR)
+      #else
         pending = size - entry.bytes_read
 
         while pending > 0 do
@@ -58,7 +58,7 @@
           raise UnexpectedEOF if @io.eof?
           pending -= bread
         end
-      end
+      #end
 
       @io.read skip # discard trailing zeros
 
